workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/src/Generation/Model/Details/DetailsStore.cs

```diff
@@ -0,0 +1,144 @@
+                .Select(component => (component.Identifier, new UnityComponentDetails(component, this)))
+                .ToDictionary(tuple => tuple.Item1, tuple => tuple.Item2);
+
+            Components = new ReadOnlyDictionary<Identifier, UnityComponentDetails>(components);
+
+            SchemaFiles = bundle.SourceMap.SourceReferences.Values
+                .Select(sourceRef => sourceRef.FilePath)
+                .Distinct()
+                .ToList()
+                .AsReadOnly();
+
+            foreach (var kv in Types)
+            {
+                kv.Value.PopulateFields(this);
+                kv.Value.PopulateChildren(this);
+            }
+
+            foreach (var kv in Components)
+            {
+                kv.Value.PopulateFields(this);
+            }
+        }
+
+        public HashSet<Identifier> GetNestedTypes(Identifier identifier)
+        {
+            return Types.Select(t => t.Key).Where(identifier.IsChild)
+                .Concat(Enums.Select(t => t.Key).Where(identifier.IsChild))
+                .ToHashSet();
+        }
+
+        private void PopulateBlittableMaps()
+        {
+            // Populate all enums.
+            foreach (var enumm in bundle.BundleContents.EnumDefinitions)
+            {
+                blittableMap.Add(enumm.EnumIdentifier, true);
+            }
+
+            var typesToTraverse = new Queue<TypeDefinitionRaw>(bundle.BundleContents.TypeDefinitions);
+
+            while (typesToTraverse.Count > 0)
+            {
+                var type = typesToTraverse.Dequeue();
+                var blittableCheckResult = CheckBlittable(type.Fields);
+
+                if (!blittableCheckResult.HasValue)
+                {
+                    typesToTraverse.Enqueue(type);
+                    continue;
+                }
+
+                blittableMap.Add(type.Identifier, blittableCheckResult.Value);
+            }
+
+            foreach (var component in bundle.BundleContents.ComponentDefinitions)
+            {
+                var blittableCheckResult = CheckBlittable(component.Fields);
+
+                if (!blittableCheckResult.HasValue)
+                {
+                    throw new InvalidOperationException("Could not check blittable-ness of component");
+                }
+
+                blittableMap.Add(component.Identifier, blittableCheckResult.Value);
+            }
+        }
+
+        private bool? CheckBlittable(IEnumerable<Field> fields)
+        {
+            foreach (var field in fields)
+            {
+                if (field.Map != null || field.List != null || field.Option != null)
+                {
+                    return false;
+                }
+
+                var innerSingularType = field.Singular.Type;
+
+                if (innerSingularType.UserType != null)
+                {
+                    if (!blittableMap.TryGetValue(Identifier.FromQualifiedName(innerSingularType.UserType.QualifiedName), out var isFieldBlittable))
```

Comment:
Why can't you use the `Identifier` directly?

Reply:
There isn't an identifier inside the type definitions:

```
     {
      "identifier": {
       "qualifiedName": "playground.SpinnerColor.color",
       "name": "color",
       "path": [
        "playground",
        "SpinnerColor",
        "color"
       ]
      },
      "fieldId": 1,
      "transient": false,
      "singularType": {
       "type": {
        "enum": {
         "qualifiedName": "playground.Color"
        }
       }
      },
      "annotations": []
     }
```

Reply:
Same issue as discussed above - will raise an FR